content/docs/hooks-overview.md

```diff
@@ -153,32 +153,32 @@ function FriendStatusWithCounter(props) {
 
->Detailed Explanation
+>Penjelasan Detail
 >
->You can learn more about `useEffect` on a dedicated page: [Using the Effect Hook](/docs/hooks-effect.html).
+>Anda dapat belajar lebih lanjut mengenai `useEffect` pada laman khusus: [Menggunakan Hook Effect](/docs/hooks-effect.html).
 
-## ✌️ Rules of Hooks {#rules-of-hooks}
+## ✌️ Peraturan Hooks {#rules-of-hooks}
 
-Hooks are JavaScript functions, but they impose two additional rules:
+*Hooks* merupakan fungsi JavaScript, tetapi memberlakukan dua aturan tambahan:
 
-* Only call Hooks **at the top level**. Don’t call Hooks inside loops, conditions, or nested functions.
-* Only call Hooks **from React function components**. Don’t call Hooks from regular JavaScript functions. (There is just one other valid place to call Hooks -- your own custom Hooks. We'll learn about them in a moment.)
+* Hanya panggil *Hooks* pada **tingkat teratas**. Jangan panggil *Hooks* di dalam perulangan, kondisi, atau fungsi bertingkat.
+* Hanya panggil *Hooks* dari **fungsi komponen React**. Jangan memanggil *Hooks* dari fungsi reguler JavaScript. (Hanya ada satu tempat yang diperbolehkan untuk memanggil *Hooks* -- *Hooks* buatan Anda sendiri. Kita akan belajar tentang hal tersebut sebentar lagi.)
 
-We provide a [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to enforce these rules automatically. We understand these rules might seem limiting or confusing at first, but they are essential to making Hooks work well.
+Kami menyediakan [*plugin linter*](https://www.npmjs.com/package/eslint-plugin-react-hooks) untuk memaksa aturan-aturan ini secara otomatis. Kami memahami aturan-aturan ini mungkin tampak membatasi atau membingungkan pada awalnya, tetapi mereka sangat penting untuk membuat *Hooks* berfungsi dengan baik.
 
->Detailed Explanation
+>Penjelasan Detail
 >
->You can learn more about these rules on a dedicated page: [Rules of Hooks](/docs/hooks-rules.html).
+>Anda dapat belajar lebih lanjut mengenai aturan-aturan tersebut pada laman khusus: [Peraturan Hooks](/docs/hooks-rules.html).
 
-## 💡 Building Your Own Hooks {#building-your-own-hooks}
+## 💡 Membangun Hooks Anda Sendiri {#building-your-own-hooks}
 
-Sometimes, we want to reuse some stateful logic between components. Traditionally, there were two popular solutions to this problem: [higher-order components](/docs/higher-order-components.html) and [render props](/docs/render-props.html). Custom Hooks let you do this, but without adding more components to your tree.
+Terkadang, kami ingin menggunakan kembali logika *stateful* antar komponen. Secara tradisional, ada dua solusi populer untuk masalah ini: [*higher-order component*](/docs/higher-order-components.html) dan [*prop render*](/docs/render-props.html). *Hooks* kustom membolehkan, tetapi tanpa menambahkan komponen lagi pada *tree*.
```

Comment:
```suggestion
Terkadang, kita ingin menggunakan kembali logika *stateful* antar komponen. Secara tradisional, ada dua solusi populer untuk masalah ini: [*higher-order component*](/docs/higher-order-components.html) dan [*render props*](/docs/render-props.html). *Hooks* kustom memungkinkan anda melakukan ini, tanpa menambahkan komponen lain pada *tree* komponen Anda.
```
*render props* is a common React terms so I don't think we need to translate it
What do you think?